### ISSUE TITLE
Controller Unit Tests failing on Chapter 15

### ISSUE THREAD

**Issue by jondevel** [wimdeblauwe/taming-thymeleaf-sources#10]
Hi,

On Chapter 15, the `testGetUsersAsUser` test fails with the following logs:

```
Caused by: java.io.FileNotFoundException: class path resource [templates/svg/users/list.svg] cannot be opened because it does not exist
	at org.springframework.core.io.ClassPathResource.getInputStream(ClassPathResource.java:215)
	at org.thymeleaf.spring6.templateresource.SpringResourceTemplateResource.reader(SpringResourceTemplateResource.java:103)
	at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parse(AbstractMarkupTemplateParser.java:223)
	... 89 more
```

I thought it could be because I've been following the book with the latest Spring Boot version however, I've also tested with the official repository code and it shows the same problem/exception.

I've come across this problem before and usually it happens when Thymeleaf can't find the template it was asked to render (from the Controller). Because of the directives/configs to process `SVG` files, the path shown in the exception looks odd, but it actually means that Thymeleaf couldn't find the `/users/list.html` template. I'm wondering if the test itself is only looking for those templates in its own `resources` folder (`/src/test/resources`) and not in the `main` (`/src/main/resources`) folder too?

**Comment by jondevel** [comment]
Oh, yeah... Just to confirm my suspicions, I've copied the templates folder to `/src/test/resources` and the test passed. So I guess the fix for this problem would be some kind of directive to tell Spring/Thymeleaf to look into `/src/main/resources/templates` as well when looking for templates.

**Comment by jondevel** [comment]
Nevermind, folks... It was just a compilation problem. I was running all the tests directly from IntelliJ IDEA counting that it'd take care of the proper compilation/packaging, but it doesn't. I realized that when I ran `mvn test` and the tests passed. So I just compiled everything manually (`mvn clean package`) and rerun the tests through IntelliJ IDEA and everything works now.

I'll close off the ticket and leave it as a reference if anyone else comes across this problem.
